src/util/loading.cpp

```diff
@@ -64,11 +64,11 @@ load_manifest() {
 
             // Will check if path exists
             GlobalContext& context = GlobalContext::instance();
-            auto future = context.submit(
-                [&object_handler](std::filesystem::path path) {
-                    object_handler.read_object(path);
-                },
-                std::move(entity_path)
+            auto future = context.submit_task(
+                [&object_handler, entity_path]() {
```

Comment:
Instead of capturing the handler, get the instance in the lambda

The less captures, the better

Also, is the object handler thread-safe?

Reply:
I think so